src/server/adapters/logger/logProvider.ts

```diff
@@ -54,15 +63,76 @@ const getLogDNA = ({ apiKey, environment }: LogDNAParams): LogProvider => {
     };
 };
 
+const getPino = ({ logFilePath, environment }: PinoParams): LogProvider => {
+    const nrPinoConf = nrPino();
+
+    const pinoConf = {
+        ...nrPinoConf,
+        formatters: {
+            ...nrPinoConf.formatters,
+            level: (label: string) => ({ level: label })
+        },
+        base: {
+            service: environment === "prod" ? "microsoft-teams-app" : `microsoft-teams-app-${environment}`
+        }
+    };
+    const stream = logFilePath ? pino.destination(logFilePath) : undefined;
+    if (stream) {
+        /*
+         This is required for pino to keep on writing logfile after log rotation.
+         'SIGHUP' event be triggered by logrotate after log rotation.
+         */
+        process.on("SIGHUP", () => stream.reopen());
+    }
+
+    const logger = stream ? pino(pinoConf, stream) : pino(pinoConf);
+
+    return {
+        info: (message, { meta }): void => logger.info?.({ meta }, message),
+        error: (message, { meta }): void => logger.error?.({ meta }, message),
+        flush: async (): Promise<void> => {
+            logger.flush();
+            await once(logger, "cleared");
+        }
+    };
+};
+
+const getMultipleLogger = ({ apiKey: logDNAApiKey, environment, logFilePath }:MultipleLogParams): LogProvider => {
+    const logdnaLogger = getLogDNA({ apiKey: logDNAApiKey, environment });
+    const pinoLogger = getPino({ environment, logFilePath });
+    return {
+        info: (message:string, { meta }): void => {
+            logdnaLogger.info(message, { meta });
+            pinoLogger.info(message, { meta });
+        },
+        error: (message:string, { meta }): void => {
+            logdnaLogger.error(message, { meta });
+            pinoLogger.error(message, { meta });
+        },
+        flush: async (): Promise<void> => {
+            await logdnaLogger.flush();
+            await pinoLogger.flush();
+        }
+    };
+};
+
 interface LogProviderGetParams {
     logDNAApiKey?: string;
     environment: string;
+    logFilePath: string;
 }
 
-const getLogProvider = ({ logDNAApiKey, environment }: LogProviderGetParams): LogProvider => {
-    if (logDNAApiKey) {
-        return getLogDNA({ apiKey: logDNAApiKey, environment });
+const getLogProvider = ({ logDNAApiKey, environment, logFilePath }: LogProviderGetParams): LogProvider => {
+    if (environment !== "local") {
+        if (logDNAApiKey && logFilePath) {
+            return getMultipleLogger({ apiKey: logDNAApiKey, environment, logFilePath });
+        } else if (logDNAApiKey) {
+            return getLogDNA({ apiKey: logDNAApiKey, environment });
+        } else if (logFilePath) {
+            return getPino({ logFilePath, environment });
+        }
     }
+
     return getConsole();
```

Comment:
Here adapters are environment agnostics. It's better not to use the environment as an if condition. Here's my suggestion which also covers @yuqu's comment:
```suggestion
    if (logDNAApiKey && logFilePath) {
        return getMultipleLogger({ apiKey: logDNAApiKey, environment, logFilePath });
    } 
    if (logDNAApiKey) {
        return getLogDNA({ apiKey: logDNAApiKey, environment });
    }
    return getPino({ logFilePath, environment });
```

Or simply this if we're ok with using pino for all cases:

```suggestion 
    if (logDNAApiKey) {
        return getMultipleLogger({ apiKey: logDNAApiKey, environment, logFilePath });
    }
    return getPino({ logFilePath, environment });
```

Reply:
Ok I've removed env check, also I added  console logger as fallback logger